crypto/kzg/src/trusted_setup.rs

```diff
@@ -0,0 +1,160 @@
+                formatter.write_str("A 48 byte hex encoded string")
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                let point = hex::decode(strip_prefix(v))
+                    .map_err(|e| de::Error::custom(format!("Failed to decode G1 point: {}", e)))?;
+                if point.len() != BYTES_PER_G1_POINT {
+                    return Err(de::Error::custom(format!(
+                        "G1 point has invalid length. Expected {} got {}",
+                        BYTES_PER_G1_POINT,
+                        point.len()
+                    )));
+                }
+                let mut res = [0; BYTES_PER_G1_POINT];
+                res.copy_from_slice(&point);
+                Ok(G1Point(res))
+            }
+        }
+
+        deserializer.deserialize_str(G1PointVisitor)
+    }
+}
+
+impl<'de> Deserialize<'de> for G2Point {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct G2PointVisitor;
+
+        impl<'de> Visitor<'de> for G2PointVisitor {
+            type Value = G2Point;
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("A 96 byte hex encoded string")
+            }
+
+            fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                let point = hex::decode(strip_prefix(v))
+                    .map_err(|e| de::Error::custom(format!("Failed to decode G2 point: {}", e)))?;
+                if point.len() != BYTES_PER_G2_POINT {
+                    return Err(de::Error::custom(format!(
+                        "G2 point has invalid length. Expected {} got {}",
+                        BYTES_PER_G2_POINT,
+                        point.len()
+                    )));
+                }
+                let mut res = [0; BYTES_PER_G2_POINT];
+                res.copy_from_slice(&point);
+                Ok(G2Point(res))
+            }
+        }
+
+        deserializer.deserialize_str(G2PointVisitor)
+    }
+}
+
+fn deserialize_g1_points<'de, D>(deserializer: D) -> Result<Vec<G1Point>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let mut decoded: Vec<G1Point> = serde::de::Deserialize::deserialize(deserializer)?;
+    // FIELD_ELEMENTS_PER_BLOB is a compile time parameter that
+    // depends on whether lighthouse is compiled with minimal or mainnet features.
+    // Minimal and mainnet trusted setup parameters differ only by the
+    // number of G1 points they contain.
+    //
+    // Hence, we truncate the number of G1 points after deserialisation
+    // to ensure that we have the right number of g1 points in the
+    // trusted setup.
+    decoded.truncate(FIELD_ELEMENTS_PER_BLOB);
```

Comment:
We seem to be always using the mainnet `FIELD_ELEMENTS_PER_BLOB` here. If it works, maybe we should alter this comment.

Reply:
You are right, this is redundant now. The actual preset defined TrustedSetup is loaded in https://github.com/sigp/lighthouse/blob/deneb-free-blobs/common/eth2_network_config/src/lib.rs#L37-L57

Gonna clean this up in a separate PR along with incorporating some ckzg improvements that were merged upstream.